todo-app/src/components/TodoContainer/TodoContainer.test.js

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { URL, fetchTodos } from "../../utils";
+import TodoContainer from "./TodoContainer";
+import axios from 'axios';
+import { RecoilRoot } from 'recoil';
+import { render} from '@testing-library/react';
+import "babel-polyfill";
+
+jest.mock("axios");
+
+describe ('<TodoContainer />', () => {
+  let component;
+  test('todos API is successfully called on page load', async() => {
+    const todos = [
+      {
+        description: "take out the trash by 4 PM wednesday",
+        id: 1,
+        status: "complete",
+        title: "take out trash"
+      }
+    ]
+    axios.get.mockResolvedValueOnce(todos);
+    const result = await fetchTodos();
+    expect(result).toEqual(todos);
+    expect(axios.get).toHaveBeenCalledWith(URL);
+  });
+
+  test('todos API fails', async() => {
+    const message = "Network Error";
+    axios.get.mockRejectedValueOnce(new Error(message));
+    const result = await fetchTodos();
+    expect(axios.get).toHaveBeenCalledWith(URL);
+    expect(result).toEqual([]);
+  });
+
+  test('renders the incomplete and complete column', () => {
+    component = render(
+        <RecoilRoot>
+          <TodoContainer />
+        </RecoilRoot>
+    )
+    const incompleteColumn = component.getAllByTestId('incomplete-column')
+    expect(incompleteColumn).toBeTruthy()
+
+    const completeColumn = component.getAllByTestId('complete-column')
+    expect(completeColumn).toBeTruthy()
+  });
```

Comment:
@danielries do these new tests accomplish what you asked for?

Reply:
I think we will need to give this test another go overall. I think we can safely remove `react-test-renderer` because everything we need to do should be accomplishable with the `@testing-library/react` library that's been used on other tests so far.

So it appears what is currently being tested here is passing a `todo` prop to `TodoContainer`. The primary issue with this is that TodoContainer does not appear to accept/use a `todo` prop. So even though we're testing it being passed in, and checking the props of the `TodoContainer`, it's not actually doing anything with that prop. So more-or-less, it's currently just testing "did the prop I passed in get passed in" and that will essentially always be true for a parent/top-level component.

What would be more robust to test is "what is different in the UI when I pass ___ prop/state information", which is effectively testing what a user would see, rather than what the application's internals see.

Since this app uses axios to pull from the API when the `TodoContainer` loads, and we already have axios mocked in this test, we should be able to reuse that pattern, and render the RecoilRoot+TodoContainer without any further state/props changes. Didn't fully test this out, but some kind of pattern like the following should work out:

```javascript
  test('renders incomplete and complete items', async () => {
    axios.get.mockResolvedValueOnce({ data: todos });

    const { getByTestId } = render(
      <RecoilRoot>
        <TodoContainer />
      </RecoilRoot>
    );

    const incomplete = getByTestId('incomplete-column');
    const complete = getByTestId('complete-column');

    const incompleteToDo = await waitFor(() => (
      within(incomplete).getByText(/go grocery shopping/)
    ));

    expect(incompleteToDo).toBeTruthy();

    const completeToDo = await waitFor(() => (
      within(complete).getByText(/take out trash/)
    ));

    expect(completeToDo).toBeTruthy();
  });
```

As you can see above, we can then check for our specific complete and incomplete todos and ensure they're being rendered. Note that we'll need to import over `within` & `waitFor` from `@testing-library/react`, but I think that should accomplish the whole "mock network request -> render todo container -> make sure the mocked todos appear in the UI" flow.